src/modules/import/api/import.controller.ts

```diff
@@ -0,0 +1,138 @@
+    UseFilters,
+    UseInterceptors,
+} from '@nestjs/common';
+import {
+    ApiBadRequestResponse,
+    ApiBearerAuth,
+    ApiConsumes,
+    ApiForbiddenResponse,
+    ApiInternalServerErrorResponse,
+    ApiNotFoundResponse,
+    ApiOAuth2,
+    ApiOkResponse,
+    ApiProduces,
+    ApiTags,
+    ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import { SchulConnexValidationErrorFilter } from '../../../shared/error/schulconnex-validation-error.filter.js';
+import { AuthenticationExceptionFilter } from '../../authentication/api/authentication-exception-filter.js';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { DbiamPersonenkontextImportBodyParams } from './dbiam-personenkontext-import.body.params.js';
+import { ImportWorkflowFactory } from '../domain/import-workflow.factory.js';
+import { PersonPermissions } from '../../authentication/domain/person-permissions.js';
+import { Permissions } from '../../authentication/api/permissions.decorator.js';
+import { DomainError } from '../../../shared/error/domain.error.js';
+import { ImportUploadResultFields, ImportWorkflowAggregate } from '../domain/import-workflow.js';
+import { DbiamImportError } from './dbiam-import.error.js';
+import { ImportvorgangByIdBodyParams } from './importvorgang-by-id.body.params.js';
+import { Response } from 'express';
+import { ImportUploadResponse } from './importvorgang-upload.response.js';
+import { ImportDomainError } from '../domain/import-domain.error.js';
+import { SchulConnexErrorMapper } from '../../../shared/error/schul-connex-error.mapper.js';
+import { ImportExceptionFilter } from './import-exception-filter.js';
+
+@UseFilters(SchulConnexValidationErrorFilter, new AuthenticationExceptionFilter(), new ImportExceptionFilter())
+@ApiTags('import')
+@ApiBearerAuth()
+@ApiOAuth2(['openid'])
+@Controller({ path: 'import' })
+export class ImportController {
+    public constructor(private readonly importWorkflowFactory: ImportWorkflowFactory) {}
+
+    @Post('upload')
+    @ApiConsumes('multipart/form-data')
+    @ApiOkResponse({ description: 'Returns an import upload response object.', type: ImportUploadResponse })
+    @ApiBadRequestResponse({ description: 'The CSV file was not valid.' })
+    @ApiUnauthorizedResponse({ description: 'Not authorized to import data with a CSV file.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to import data with a CSV file.' })
+    @ApiInternalServerErrorResponse({ description: 'Internal server error while importing data with a CSV file.' })
+    @UseInterceptors(FileInterceptor('file'))
+    public async uploadFile(
+        @Body() body: DbiamPersonenkontextImportBodyParams,
+        @UploadedFile(
+            new ParseFilePipeBuilder()
+                .addFileTypeValidator({
+                    fileType: 'text/csv',
+                })
+                .build({
+                    errorHttpStatusCode: HttpStatus.BAD_REQUEST,
+                    fileIsRequired: false,
```

Comment:
Shouldn't the file be required?

Reply:
The file is already required in the api properties, the flag does not bring much value but i set it to true.